sklearn/manifold/tests/test_spectral_embedding.py

```diff
@@ -117,33 +125,59 @@ def test_spectral_embedding_two_components(seed=36):
     true_label[0:n_sample] = 1
 
     se_precomp = SpectralEmbedding(
-        n_components=1, affinity="precomputed", random_state=np.random.RandomState(seed)
+        n_components=1,
+        affinity="precomputed",
+        random_state=np.random.RandomState(seed),
+        eigen_solver=eigen_solver,
     )
-    embedded_coordinate = se_precomp.fit_transform(affinity)
-    # Some numpy versions are touchy with types
-    embedded_coordinate = se_precomp.fit_transform(affinity.astype(np.float32))
-    # thresholding on the first components using 0.
-    label_ = np.array(embedded_coordinate.ravel() < 0, dtype="float")
-    assert normalized_mutual_info_score(true_label, label_) == pytest.approx(1.0)
+    for dtype in [np.float32, np.float64]:
+        if eigen_solver == "amg" and not amg_loaded:
```

Comment:
I think that we can isolate the testing regarding the erroring of PyAMG not being available.
Like this, we could avoid the indentation with the for loop (with a parametrization) and skip if PyAMG is not installed.

Reply:
So we can add the following test:

```python
@pytest.mark.skipif(
    not is_pyamg_not_available,
    reason="PyAMG is installed and we should not test for an error."
)
def test_error_pyamg_not_available():
    se_precomp = SpectralEmbedding(
        n_components=2,
        affinity="rbf",
        eigen_solver="amg",
    )
    err_msg = "The eigen_solver was set to 'amg', but pyamg is not available."
    with pytest.raises(ValueError, match=err_msg):
        se_precomp.fit_transform(S)
```

Reply:
@glemaitre My check also tested that there is no error if PyAMG is installed. 
Anyway, it's not related to this PR - maybe should be done in a different PR?

Reply:
> My check also tested that there is no error if PyAMG is installed.

Basically, we already have other tests that check some normal behaviour when `amg` is installed. So there is no need to do anything when `amg` is installed in this test.